moveit_ros/planning/moveit_cpp/src/planning_component.cpp

```diff
@@ -419,4 +419,25 @@ const planning_interface::MotionPlanResponse& PlanningComponent::getLastMotionPl
 {
   return last_plan_solution_;
 }
+
+bool PlanningComponent::stopAtFirstSolution(
+    PlanSolutions const& plan_solutions,
+    PlanningComponent::MultiPipelinePlanRequestParameters const& /*plan_request_parameters*/)
+{
+  // Read solutions that are found up to this point from a thread safe storage
+  auto const& solutions = plan_solutions.getSolutions();
+
+  // Stop at the first successful plan
+  for (auto const& solution : solutions)
+  {
+    if (bool(solution))
```

Comment:
Nit: You could clarify with a small comment here that `bool(solution)` evaluate the error code of the solution and is true if it is `SUCCESS`